packages/glimmer-test-helpers/lib/environment.ts

```diff
@@ -96,17 +96,53 @@ class ArrayIterator implements OpaqueIterator {
     return this.array.length === 0;
   }
 
-  next(): IterationItem<Opaque> {
+  next(): IterationItem<Opaque, number> {
     let { position, array, keyFor } = this;
 
     if (position >= array.length) return null;
 
     let value = array[position];
     let key = keyFor(value, position);
+    let memo = position;
 
     this.position++;
 
-    return { key, value };
+    return { key, value, memo };
+  }
+}
+
+type KVPair = {
+  key: string;
+  value: Opaque;
+};
```

Comment:
You could write

```typescript
interface KVPair {
  key: string;
  value: Opaque;
}
```

(Or the iterator could just take an array of keys and an array of values, since it's all internal it doesn't really matter)